### ISSUE TITLE
About the results for other datasets

### ISSUE THREAD

**Issue by LIANGKE23** [IBCNServices/pyRDF2Vec#89]
## ❓ Question

I don't know how to reproduce the results for AIFB, BGS with current version of pyRDF2VEC, since we cannot get the OWL file for these datasets. 
Can you offer a way for this?

Thanks!

**Comment by bsteenwi** [comment]
The original AIFB, MUTAG, AM and BGS datasets can be found at http://data.dws.informatik.uni-mannheim.de/rmlod/LOD_ML_Datasets/data/datasets/RDF_Datasets/

**Comment by LIANGKE23** [comment]
> The original AIFB, MUTAG, AM and BGS datasets can be found at http://data.dws.informatik.uni-mannheim.de/rmlod/LOD_ML_Datasets/data/datasets/RDF_Datasets/

I have run the mutag.py in examples folder. But when it comes to the other datasets, which way for KG initialization I can choose? (There is no OWL file in other datasets.)

**Comment by bsteenwi** [comment]
We support RDF/XML, N3, NTriples, N-Quads, Turtle, TriX, JSON-LD and RDFa files as we rely in rdflib. For AIFB, you can simply use the fixed_complete.nt file. For BGS, you can use the Complete dataset.nt

**Comment by LIANGKE23** [comment]
> We support RDF/XML, N3, NTriples, N-Quads, Turtle, TriX, JSON-LD and RDFa files as we rely in rdflib. For AIFB, you can simply use the fixed_complete.nt file. For BGS, you can use the Complete dataset.nt

Thanks for your answer!
But how can I get the skip_predicates, and  literals for these datasets? I am a little bit confused about them.

**Comment by GillesVandewiele** [comment]
Here are the label_predicates for each dataset and the columns in the associated CSV/TSVs where you can find entity and labels:
```
labels = {"AIFB" : (["http://swrc.ontoware.org/ontology#affiliation",
                     "http://swrc.ontoware.org/ontology#employs",
                     "http://swrc.ontoware.org/ontology#carriedOutBy"], "person", "label_affiliation"),
          "AM" : (["http://purl.org/collections/nl/am/objectCategory",
                   "http://purl.org/collections/nl/am/material"], "proxy", "label_cateogory"),
          "BGS": (["http://data.bgs.ac.uk/ref/Lexicon/hasLithogenesis",
                   "http://data.bgs.ac.uk/ref/Lexicon/hasLithogenesisDescription",
                   "http://data.bgs.ac.uk/ref/Lexicon/hasTheme"], "rock", "label_lithogenesis"),
          "MUTAG": (["http://dl-learner.org/carcinogenesis#isMutagenic"], "bond", "label_mutagenic")}
```

**Comment by LIANGKE23** [comment]
> We support RDF/XML, N3, NTriples, N-Quads, Turtle, TriX, JSON-LD and RDFa files as we rely in rdflib. For AIFB, you can simply use the fixed_complete.nt file. For BGS, you can use the Complete dataset.nt

How about AM? Which file I can use like nt, n3 ....? I cannot find that kind of file~

**Comment by bsteenwi** [comment]
Turtle (ttl) --> http://data.dws.informatik.uni-mannheim.de/rmlod/LOD_ML_Datasets/data/datasets/RDF_Datasets/AM/data/

**Comment by LIANGKE23** [comment]
> Turtle (ttl) --> http://data.dws.informatik.uni-mannheim.de/rmlod/LOD_ML_Datasets/data/datasets/RDF_Datasets/AM/data/

I have tried AM dataset....

The accuracy is only 34% for classification, which is 87% in RGCN paper....

Do you know how to reproduce that results with your code~?

**Comment by bsteenwi** [comment]
Can you send us your python script?

**Comment by LIANGKE23** [comment]
> Can you send us your python script?

Ok
Here is the script:

import matplotlib.pyplot as plt
import pandas as pd
from sklearn.manifold import TSNE
from sklearn.metrics import accuracy_score, confusion_matrix
from sklearn.model_selection import GridSearchCV
from sklearn.svm import SVC

from pyrdf2vec import RDF2VecTransformer
from pyrdf2vec.embedders import Word2Vec
from pyrdf2vec.graphs import KG
from pyrdf2vec.samplers import WideSampler
from pyrdf2vec.walkers import HALKWalker

# Ensure the determinism of this script by initializing a pseudo-random number.
RANDOM_STATE = 22

test_data = pd.read_csv("../samples/am/testSet.tsv", sep="\t")
train_data = pd.read_csv("../samples/am/trainingSet.tsv", sep="\t")

train_entities = [entity for entity in train_data["proxy"]]
train_labels = list(train_data["label_cateogory"])

test_entities = [entity for entity in test_data["proxy"]]
test_labels = list(test_data["label_cateogory"])

entities = train_entities + test_entities
labels = train_labels + test_labels

embeddings, literals = RDF2VecTransformer(
    # Ensure random determinism for Word2Vec.
    # Must be used with PYTHONHASHSEED.
    Word2Vec(workers=0, epochs=10),
    # Extract all walks with a maximum depth of 2 for each entity using two
    # processes and use a random state to ensure that the same walks are
    # generated for the entities without hashing as MUTAG is a short KG.
    walkers=[
        HALKWalker(
            2,
            None,
            n_jobs=2,
            sampler=WideSampler(),
            random_state=RANDOM_STATE,
            md5_bytes=None,
        )
    ],
    verbose=1,
).fit_transform(
    KG(
       "../samples/am/rdf_am-data.ttl",
        # "../samples/am/am_stripped.nt",
        skip_predicates={"http://purl.org/collections/nl/am/material",
                         "http://purl.org/collections/nl/am/objectCategory"},
        literals=[
            [
                "http://purl.org/collections/nl/am/t-14592",
                "http://purl.org/collections/nl/am/t-15459",
                "http://purl.org/collections/nl/am/t-15579",
                "http://purl.org/collections/nl/am/t-15606",
                "http://purl.org/collections/nl/am/t-22503",
                "http://purl.org/collections/nl/am/t-22504",
                "http://purl.org/collections/nl/am/t-22505",
                "http://purl.org/collections/nl/am/t-22506",
                "http://purl.org/collections/nl/am/t-22507",
                "http://purl.org/collections/nl/am/t-22508",
                "http://purl.org/collections/nl/am/t-5504",
            ],
        ],
    ),
    entities,
)

train_embeddings = embeddings[: len(train_entities)]
test_embeddings = embeddings[len(train_entities) :]

# Fit a Support Vector Machine on train embeddings and pick the best
# C-parameters (regularization strength).
clf = GridSearchCV(
    SVC(random_state=RANDOM_STATE), {"C": [10**i for i in range(-3, 4)]}
)
clf.fit(train_embeddings, train_labels)

# Evaluate the Support Vector Machine on test embeddings.
predictions = clf.predict(test_embeddings)
print(
    f"Predicted {len(test_entities)} entities with an accuracy of "
    + f"{accuracy_score(test_labels, predictions) * 100 :.4f}%"
)
print("Confusion Matrix ([[TN, FP], [FN, TP]]):")
print(confusion_matrix(test_labels, predictions))

**Comment by bsteenwi** [comment]
Based on the R-GCN paper:
```
Hyperparameters for baselines are chosen according to the best model performance in (Ristoski and Paulheim 2016), i.e.  RDF2Vec: 
2 (WL tree depth), 
4 (WL iterations), 
500 (embedding size), 
5 (window size), 
10 (SkipGram iterations), 
25 (number of negative samples). 
We optimize the SVM regularization constant C ∈ {0.001, 0.01, 0.1, 1, 10, 100, 1000} based on performance on a 80/20 train/validation split (of the original training set).
```
So you could probably use the WLWalker to obtain better results, but keep in mind that we made some remarks to the WL on RDF data in the past (see https://arxiv.org/pdf/2009.04404.pdf).
